/*
 * Copyright 2021 Google LLC.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package yggdrasil_decision_forests.distribute.proto;

import "yggdrasil_decision_forests/utils/distribute/distribute.proto";

// In process, non-parallelized distribution. For debugging and pipeline
// development.
message MultiThreadImp {
  // Number of workers / threads.
  optional int32 num_workers = 1 [default = 10];
}

extend Config {
  optional MultiThreadImp multi_thread = 1000;
}
